src/modifiercontext.ts

```diff
@@ -0,0 +1,178 @@
+// [VexFlow](http://vexflow.com) - Copyright (c) Mohit Muthanna 2010.
+//
+// ## Description
+//
+// This class implements various types of members to notes (e.g. bends,
+// fingering positions etc.)
+
+import { Vex } from './vex';
+import { StaveNote } from './stavenote';
+import { Dot } from './dot';
+import { FretHandFinger } from './frethandfinger';
+import { Accidental } from './accidental';
+import { NoteSubGroup } from './notesubgroup';
+import { GraceNoteGroup } from './gracenotegroup';
+import { Stroke } from './strokes';
+import { StringNumber } from './stringnumber';
+import { Articulation } from './articulation';
+import { Ornament } from './ornament';
+import { Annotation } from './annotation';
+import { ChordSymbol } from './chordsymbol';
+import { Bend } from './bend';
+import { Vibrato } from './vibrato';
+import { Modifier } from './modifier';
+import { TabNote } from './tabnote';
+
+export interface ModifierContextState {
+  right_shift: number;
+  left_shift: number;
+  text_line: number;
+  top_text_line: number;
+}
+
+export interface ModifierContextMetrics {
+  width: number;
+  spacing: number;
+}
+
+export type PreformatModifierType =
+  | typeof StaveNote
+  | typeof Dot
+  | typeof FretHandFinger
+  | typeof Accidental
+  | typeof Stroke
+  | typeof GraceNoteGroup
+  | typeof NoteSubGroup
+  | typeof StringNumber
+  | typeof Articulation
+  | typeof Ornament
+  | typeof Annotation
+  | typeof ChordSymbol
+  | typeof Bend
+  | typeof Vibrato;
+
+export type PostformatModifierType = typeof StaveNote;
```

Comment:
This also can be `any` and moved inline.

Reply:
done